libs/src/services/hedgehog/index.js

```diff
@@ -86,6 +86,48 @@ class HedgehogWrapper {
       }
     }
 
+    /**
+     * Generate secure credentials to allow login
+     * @param {String} username username
+     */
+    hedgehog.getEntropyFromLocalStorage = async () => {
+      let entropy = await WalletManager.getEntropyFromLocalStorage()
+      if (entropy == null) {
+        throw new Error('GenerateRecoveryInfo | Entropy missing')
+      }
+      return entropy
+    }
+
+    /**
+     * Generate new set of auth credentials to allow login
+     * @param {String} username - username
+     * @param {String} password - new password
+     * @param {String} entropy - stored entropy val
+     */
+    hedgehog.resetPassword = async (username, password, entropy) => {
+      let self = this
+      const createWalletPromise = WalletManager.createWalletObj(password, entropy)
+      const lookupKeyPromise = WalletManager.createAuthLookupKey(username, password)
+      try {
+        let result = await Promise.all([createWalletPromise, lookupKeyPromise])
+
+        const { ivHex, cipherTextHex, walletObj } = result[0]
+        const lookupKey = result[1]
+
+        const authData = {
+          iv: ivHex,
+          cipherText: cipherTextHex,
+          lookupKey: lookupKey
+        }
+
+        await self.setAuthFn(authData)
```

Comment:
So this right now just creates a new auth entry. I like this pattern. We could even optionally have another endpoint we add for like self.removeOldAuthEntry()